spec/models/user_spec.rb

```diff
@@ -8,9 +8,25 @@
     it { should validate_presence_of(col) }
   end
 
+  it { should validate_length_of(:username).is_at_least(MIN_USERNAME_LENGTH)}
+  it { should validate_length_of(:username).is_at_most(MAX_USERNAME_LENGTH)}
+  it { should have_secure_password }
+  VALID_USERNAME_ARRAY.each do |username|
```

Comment:
"username" seems a misleading variable name here. What are each of these VALID_USERNAME_ARRAY items? Not a username, but valid_characters (or what)?

Reply:
Nope, they are different usernames that all happen to be valid (e.g. No special characters, varying lengths all within 6-20). We also made an INVALID array.